### PR TITLE
[HTP-130] YOC: Added Trade Desk Unified ID support

### PR DESCRIPTION
**Pull request by mk0x9**
## Type of Change

- [ ] New adapter
- [x] Feature Update
- [ ] Bug Fix
- [ ] Code Refactor
- [ ] Other

## Description of Change
Added support for Trade Desk Unified ID & GDPR parameters to user sync requests

### CODE REVIEW

**Comment by ix-certification** [comment]
There is no JIRA ticket created for this pull request to be reviewed.

**Comment by mk0x9** [comment]
@ix-certification here is it: https://jira.indexexchange.com/servicedesk/customer/portal/15/HTP-130

File: yoc/yoc-htb.js [hunk 1]
```diff
@@ -342,6 +359,15 @@ function YocHtb(configs) {
             var pixelUrl = Browser.getProtocol() + '//t.visx.net/push_sync?wrapperType=IX&wrapperVersion='
                 + SpaceCamp.version + '&adapterVersion=' + __profile.version;
 
+            var gdprStatus = ComplianceService.gdpr.getConsent();
+            if (gdprStatus) {
+                if (gdprStatus.consentString) {
+                    pixelUrl = pixelUrl + '&gdpr_consent=' + encodeURIComponent(gdprStatus.consentString);
+                }
+                // eslint-disable-next-line camelcase
+                pixelUrl = pixelUrl + '&gdpr_applies=' + Utilities.isBoolean(gdprStatus.applies) ? Number(gdprStatus.applies) : 1;
```
Comment (ix-certification) [inline yoc/yoc-htb.js:368]: Linting is failing on line 368: `maximum line length of 120 max-len`

File: yoc/yoc-htb.js [hunk 2]
```diff
@@ -342,6 +359,15 @@ function YocHtb(configs) {
             var pixelUrl = Browser.getProtocol() + '//t.visx.net/push_sync?wrapperType=IX&wrapperVersion='
                 + SpaceCamp.version + '&adapterVersion=' + __profile.version;
 
+            var gdprStatus = ComplianceService.gdpr.getConsent();
```
Comment (ix-certification) [inline yoc/yoc-htb.js:362]: Missing system tests to validate request contains `gdpr` data. Please refer to the following doc for more information: https://kb.indexexchange.com/ix_library_partners/build_adapter/test_case_reference.htm#validateBidRequestWithTcf2

File: yoc/yoc-htb.js [hunk 3]
```diff
@@ -178,6 +182,19 @@ function YocHtb(configs) {
         queryObj.cb
             = 'window.' + SpaceCamp.NAMESPACE + '.' + __profile.namespace + '.adResponseCallbacks.' + callbackId;
 
+        if (identityData && identityData.AdserverOrgIp && identityData.AdserverOrgIp.data) {
+            var adsrvrUids = identityData.AdserverOrgIp.data.uids;
+            if (Utilities.isArray(adsrvrUids)) {
+                for (var j = 0; j < adsrvrUids.length; j++) {
+                    if (adsrvrUids[j].ext && adsrvrUids[j].ext.rtiPartner === 'TDID') {
+                        queryObj.tdid = adsrvrUids[j].id;
```
Comment (ix-certification) [inline yoc/yoc-htb.js:190]: Missing system tests to validate request contains `AdserverOrgIp` data. Please refer to the following doc for more information: https://kb.indexexchange.com/ix_library_partners/build_adapter/test_case_reference.htm#validateBidRequestWithUAdSrvrOrg

**Review by ix-certification — COMMENTED** [review]
Hi YOC Team,
Please see the comments from the code review below. Would you also be able to confirm if these changes are intended to go into `canary` first or directly to `production`? Right now, this pull request is for production.

File: yoc/yoc-htb.js (hunk 3 above)
Comment (mk0x9) [reply yoc/yoc-htb.js:190]: Test has been added

Comment (mk0x9) [reply yoc/yoc-htb.js:190]: Test had been added

File: yoc/yoc-htb.js (hunk 2 above)
Comment (mk0x9) [reply yoc/yoc-htb.js:362]: Test had been added

**Pull request closed without merging**